core/src/main/scala-3/cr/pulsar/data.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Chatroulette
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cr.pulsar
+
+import scala.concurrent.duration.FiniteDuration
+import scala.util.matching.Regex
+
+object data:
+  val OperationTimeout = NewType.of[FiniteDuration]
+  type OperationTimeout = OperationTimeout.Type
+
+  val ConnectionTimeout = NewType.of[FiniteDuration]
+  type ConnectionTimeout = ConnectionTimeout.Type
+
+  val PulsarTenant = NewType.of[String]
+  type PulsarTenant = PulsarTenant.Type
+
+  val PulsarNamespace = NewType.of[String]
+  type PulsarNamespace = PulsarNamespace.Type
+
+  val PulsarURL = NewType.of[String]
+  type PulsarURL = PulsarURL.Type
+
+  val SubscriptionName = NewType.of[String]
+  type SubscriptionName = SubscriptionName.Type
+
+  val TopicName = NewType.of[String]
+  type TopicName = TopicName.Type
+
+  val TopicNamePattern = NewType.of[Regex]
+  type TopicNamePattern = TopicNamePattern.Type
+
+  val TopicURL = NewType.of[String]
+  type TopicURL = TopicURL.Type
+
+  object context:
+    final case class Tenant(value: String)
+    final case class Namespace(value: String)
+    final case class FunctionName(value: String)
+    final case class FunctionId(value: String)
+    final case class InstanceId(value: Int)
+    final case class NumInstances(value: Int)
+    final case class FunctionVersion(value: String)
+    final case class InputTopic(value: String)
+    final case class OutputTopic(value: String)
+    final case class OutputSchemaType(value: String)
+  end context
```

Comment:
Why do you prefer to keep `end` statements?

Reply:
Welcome to Scala 3 where the same things can be written in 100 forms! :roll_eyes: 

Are you suggesting I change it to curly braces and use a 3-space indentation like Martin suggests? :laughing:

Reply:
True 😄 

I think it should be possible just to drop them

Reply:
In all fairness, this was just a syntax playground since it's the only Scala 3 file we have in this repo so I'm not married to any style and I can change it back to our Scala 2 style if preferred.

Reply:
Obviously, it's not a big deal, just looks very old school 😄

Reply:
Personally I've found `end` to be very useful when you have more nesting and longer blocks - e.g. https://github.com/kubukoz/slang/blob/0a2841e322b0b6c8dad7e87ebbc354c052670db8/src/main/scala/com/kubukoz/slang/qualifier/Qualifier.scala#L127 - can recommend that approach